src/libpgagroal/pipeline_transaction.c

```diff
@@ -144,13 +144,10 @@ transaction_start(struct ev_loop* loop, struct worker_io* w)
    if (is_new)
    {
       /* Sleep for 5ms */
-      struct timespec ts;
-      ts.tv_sec = 0;
-      ts.tv_nsec = 5000000L;
-      nanosleep(&ts, NULL);
-   }
+      SLEEP(5000000L)
 
-   return;
+      return;
```

Comment:
You moved the `return;` statement under the `if`

Reply:
It took me some time to understand how did I mess such branch.
It turned out it was `uncrustify` that added the extra curly braces after I wrote the code as:

```
if (is_new)
   SLEEP(...)

return;
```

Since I run uncrusityf as the last thing before pushing, I did not noticed the code changed to:

```
if (is_new)
{
   SLEEP(..)
  return;
}
```

Commit 6a62e4b6a0a62d68df62ec887430424a8641a3f6 fixes this by adding an extrac curly braces pair to the `if`.
I'm not an expert in using `uncrustify` but probably we should document this or fix (if it can understand it is a macro, I suspect the absence of `;` is causing the problem).